packages/bitcore-wallet-client/src/lib/key.ts

```diff
@@ -278,25 +278,12 @@ export class Key {
     $.checkArgument(opts.n >= 1, 'n need to be >=1');
 
     let purpose = opts.n == 1 || this.use44forMultisig ? '44' : '48';
-    var coinCode = '0';
-
-    if (opts.network == 'testnet' && opts.coin !== 'eth') {
-      coinCode = '1';
-    } else if (opts.coin == 'bch') {
-      if (this.use0forBCH) {
-        coinCode = '0';
-      } else {
-        coinCode = '145';
-      }
-    } else if (opts.coin == 'btc') {
-      coinCode = '0';
-    } else if (opts.coin == 'eth') {
-      coinCode = '60';
-    } else {
-      throw new Error('unknown coin: ' + opts.coin);
+    let coin = opts.coin;
+    const network = opts.network || 'livenet';
+    if (this.use0forBCH && opts.coin === 'bch') {
```

Comment:
si comment before. I would prefer you send an extra `opts` in pathFor with the flag: `use0forBCH` or something like that.